### PR TITLE
Feature/create hasura service

### PR DESCRIPTION
**Pull request by ismetkizgin**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,7 +84,7 @@
   },
   "homepage": "https://github.com/BrewInteractive/nestjs-hasura#readme-module",
   "dependencies": {
-    "graphql-request": "^7.0.1"
+    "graphql-request": "^6.1.0"
```
Comment (mfozmen) [inline package.json:87]: Why older version?

File: src/hasura.service.spec.ts [hunk 2]
```diff
@@ -1,18 +1,139 @@
...
+      authorization: hasuraOptions.authorization,
+      'x-hasura-use-backend-only-permissions':
+        hasuraOptions.useBackendOnlyPermissions,
+      'x-hasura-admin-secret': hasuraConfig.adminSecret,
+    });
   });
 
-  it('should be defined', () => {
-    expect(service).toBeDefined();
+  it('should throw error if there is no admin secret ', async () => {
```
Comment (mfozmen) [inline src/hasura.service.spec.ts]: It must be run with or without admin secret. This test description is invalid.

File: src/models/hasura-options.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export class HasuraOptions {
+  useBackendOnlyPermissions?: boolean;
+  useAdminSecret?: boolean;
+  authorization?: string;
```
Comment (mfozmen) [inline src/models/hasura-options.ts]: We don't need this HasuraOptions class. We can move this to HasuraRequestOptions.

Comment (mfozmen) [inline src/models/hasura-options.ts]: Rename authorization as authorization token.

File: src/models/hasura-options-flags.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export enum HasuraOptionsFlags {
+  useAdminSecret = 1, // 0001
+  useBackendOnlyPermissions = 3, // 0011
+  role = 4, // 0100
```
Comment (mfozmen) [inline src/models/hasura-options-flags.ts]: role and authorization is not a feature flag.

File: src/hasura.service.ts [hunk 5]
```diff
@@ -1,4 +1,92 @@
...
+    string
+  > = {
+    [HasuraOptionsFlags.authorization]: 'authorization',
+    [HasuraOptionsFlags.useBackendOnlyPermissions]:
+      'x-hasura-use-backend-only-permissions',
+    [HasuraOptionsFlags.role]: 'x-hasura-role',
+    [HasuraOptionsFlags.useAdminSecret]: 'x-hasura-admin-secret',
+  };
+
+  private getHasuraOptionsFlags() {
```
Comment (mfozmen) [inline src/hasura.service.ts]: Why are we converting again? We can make just one enum or class them.

**Review by mfozmen — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (ismetkizgin) [reply package.json:87]: Downgraded due to a bug in the new version.

File: src/hasura.service.ts [hunk 6]
```diff
@@ -1,4 +1,92 @@
...
+  HasuraOptionsFlags,
+  HasuraRequestOptions,
+} from './models';
+
 import { Injectable } from '@nestjs/common';
 
 @Injectable()
-export class HasuraService {}
+export class HasuraService {
+  private readonly _graphQLClient: GraphQLClient;
```
Comment (mfozmen) [inline src/hasura.service.ts]: do not start with _ as a prefix in private variables since this is a typescript.

**Review by mfozmen — APPROVED** [review]

**Review by mfozmen — APPROVED** [review]

**Pull request merged**